pom.xml

```diff
@@ -921,104 +920,6 @@
-                        <artifactId>maven-javadoc-plugin</artifactId>
-                        <version>${maven.javadoc.plugin.version}</version>
-                        <configuration>
-                            <maxmemory>1024</maxmemory>
-                            <excludePackageNames>${maven.javadoc.plugin.excludePackageNames}</excludePackageNames>
-                            <doclint>-missing</doclint>
-                        </configuration>
-                        <executions>
-                            <execution>
-                                <id>attach-javadocs</id>
-                                <goals>
-                                    <goal>jar</goal>
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-
-                    <plugin>
-                        <groupId>org.sonatype.plugins</groupId>
-                        <artifactId>nexus-staging-maven-plugin</artifactId>
-                        <version>1.6.8</version>
-                        <extensions>true</extensions>
-                        <configuration>
-                            <serverId>release-repository</serverId>
-                            <nexusUrl>https://oss.sonatype.org/</nexusUrl>
-                            <autoReleaseAfterClose>true</autoReleaseAfterClose>
-                        </configuration>
-                    </plugin>
-                </plugins>
-            </build>
-            <modules>
-                <module>hazelcast-all</module>
-            </modules>
-        </profile>
-
-        <!-- unfortunately it isn't possible to chain profiles, so we need to duplicate the javadoc plugin -->
-        <profile>
-            <id>release-snapshot</id>
-            <properties>
-                <javadoc>true</javadoc>
-            </properties>
-            <modules>
-                <module>hazelcast-all</module>
-            </modules>
-            <build>
-                <plugins>
-                    <plugin>
-                        <groupId>org.apache.maven.plugins</groupId>
-                        <artifactId>maven-javadoc-plugin</artifactId>
-                        <version>${maven.javadoc.plugin.version}</version>
-                        <configuration>
-                            <excludePackageNames>${maven.javadoc.plugin.excludePackageNames}</excludePackageNames>
-                            <doclint>-missing</doclint>
-                        </configuration>
-                        <executions>
-                            <execution>
-                                <id>attach-javadocs</id>
-                                <goals>
-                                    <goal>jar</goal>
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
```

Comment:
AFAIK, we still need the `release` and `release-snapshot` profiles for the deployments. CC @olukas Could you confirm? Not sure about the `zip` one.

Reply:
Yes, I'd say at least `release`. He did mention it on release night as something he was running - https://hazelcast.slack.com/archives/C026A3BMTQD/p1626204511408900?thread_ts=1626204309.408000&cid=C026A3BMTQD

Reply:
Regarding `release-snapshot`, I don't see much reference to it on slack (which doesn't mean it's not used). Only this - https://github.com/hazelcast/hazelcast/issues/14989

Reply:
The profile is used in deploy jobs (from dev branches). E.g. http://jenkins.hazelcast.com/job/Hazelcast-master-deploy/configure